app/src/main/java/org/hackillinois/android/view/navigationdrawer/ScannerFragment.kt

```diff
@@ -0,0 +1,110 @@
+package org.hackillinois.android.view.navigationdrawer
+
+import android.Manifest
+import android.R
+import android.arch.lifecycle.Observer
+import android.arch.lifecycle.ViewModelProviders
+import android.content.pm.PackageManager
+import android.os.Build
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
+import android.widget.Toast
+import com.google.zxing.ResultPoint
+import com.journeyapps.barcodescanner.BarcodeCallback
+import com.journeyapps.barcodescanner.BarcodeResult
+import kotlinx.android.synthetic.main.fragment_scanner.*
+import org.hackillinois.android.viewmodel.ScannerViewModel
+
+
+class ScannerFragment : Fragment() {
+    var lastScannedUser: String = ""
+    val PERMISSIONS_REQUEST_ACCESS_CAMERA = 0
+    lateinit var viewModel: ScannerViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        viewModel = ViewModelProviders.of(this).get(ScannerViewModel::class.java)
+        viewModel.eventsListLiveData.observe(this, Observer { updateEventList(it) })
+
+        // Ensure the Camera permission is granted
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
+                && activity!!.checkSelfPermission(Manifest.permission.CAMERA)
+                != PackageManager.PERMISSION_GRANTED) {
+            requestPermissions(arrayOf(Manifest.permission.CAMERA),
+                    PERMISSIONS_REQUEST_ACCESS_CAMERA)
+        } else {
+            qrScanner.decodeContinuous(onQrCodeScan)
```

Comment:
I think `onCreate()` is called before `onCreateView()`, so you may have to do this in `onCreateView()` for `qrScanner` to be non-null.